### DISCUSSION TITLE
drawer width only accept a number as px unit

### DISCUSSION THREAD

**Discussion by amir-armantaheri** [quasarframework/quasar#16502] [Ideas / Proposals]
why there is no option to pass drawer width as a percentage value for example 80% 
i need a dynamic width for different screens.
for now i have handled it by Js and resize event but CSS is a better solution.
it also can accept a function which pass current state of drawer as payload and return a string which can be a px width or a percentage.
this function can be useful for changing your logic in different state of drawer for example i can return 80% percent in desktop mode and 100% in mobile mode.

**Comment by pdanpdan** [comment]
Make a codepen

**Comment by shomalikhashayar** [comment]
Howdy Amir. Try this:

:width="$q.screen.gt.xs ? 350 : deviceWidth"

const deviceWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
